### ISSUE TITLE
Profile Pictures shown to not authorized users 

### ISSUE THREAD

**Issue by k4r7u** [humhub/humhub#1096]
Unfortunately the profile images are shown at <hostname>/uploads/profile_image/<random chars>.jpg even when not logged in.
Could you provide a possibility to let only logged in users view the pictures?

I don't want everyone who knows me domain name to be able to see my profile picture.

**Comment by WASasquatch** [comment]
Have these users set their profiles to private?

**Comment by k4r7u** [comment]
Yes, the profiles are set to private. For public profiles this would be ok with me, but not for private ones.

**Comment by programguy** [comment]
These pictures are left public so that they can be shown in notification emails. If you want these to be private, you must find a new way to get the profile pictures in the emails.

**Comment by k4r7u** [comment]
I understood the reason why the pictures are public. Never the less I think this is a privacy issue. I would rather use some generic profile picture, then have all pictures exposed to the public.

**Comment by rondonjon** [comment]
Wait, does that mean I won't see any profile pictures if the HumHub server is running within a local network and I check my e-mail outside that network?

Then I would suggest to add them either inline (as base64 images) or as image attachments to the emails. Admittedly, that will increase the size of every e-mail, but with two or three image attachments, the size could still be around ~20 kb, in other words, "nothing" with regard to a standard inbox size today.

**Comment by Ace2213** [comment]
I don't think there is a way around this. Profile pictures must always be public. It would be better practice to communicate this to your users so they can avoid adding profile pictures that they wouldn't want to be public. 

That said, you can manually configure your robots.txt file as I have to prevent crawlers from scraping your public images. This won't stop them from being directly requested, but it will stop them from showing up in search results

**Comment by samsheridan** [comment]
I think there is a way around it. Profile profile pictures and other downloads could be routed through PHP (just like pretty links) where an access check is performed before the contents of the file are sent.

To make it show up in emails, profile small thumbnails could easily be included inline like @rondonjon suggested. If that isn't good enough, long unique access URLs could be generated and stored in the database for some period of time before expiring and being deleted. And if it's expired the default profile image could be sent instead as a fail safe.

**Comment by Ace2213** [comment]
I just don't think all of that is necessary for profile pictures, which everyone should know and treat as always public.

**Comment by funkycram** [comment]
Yes, profile private images (profile or related to private content) should be stored in the `protected` folder and accessible through PHP as @samsheridan suggest it.

An other enhancement, would be for public networks but with the option checked "Limited access to non identified users" (in `/admin/authentication`) : when not logged in (guest mode), profile image should be shown small (35x35 px) and not 150x150 px but reduced with CSS. Because search engines index the images in the big size...
